common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -2167,6 +2167,69 @@ object CelebornConf extends Logging {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("30s")
 
+  val MASTER_HTTP_AUTH_SUPPORTED_SCHEMES: ConfigEntry[Seq[String]] =
+    buildConf("celeborn.master.http.auth.supportedSchemes")
+      .categories("master")
+      .version("0.5.0")
+      .doc("A comma-separated list of master http auth supported schemes." +
+        "<ul>" +
+        " <li>SPNEGO: Kerberos/GSSAPI authentication.</li>" +
+        " <li>BASIC: User-defined password authentication, anonymous by default.</li>" +
+        " <li>BEARER: User-defined bearer token authentication, anonymous by default.</li>" +
+        "</ul>")
+      .stringConf
+      .toSequence
+      .createWithDefault(Nil)
+
+  val MASTER_HTTP_SPNEGO_KEYTAB: OptionalConfigEntry[String] =
+    buildConf("celeborn.master.http.spnego.keytab")
+      .categories("master")
+      .version("0.5.0")
+      .doc("The keytab file for SPNego authentication.")
+      .stringConf
+      .createOptional
+
+  val MASTER_HTTP_SPNEGO_PRINCIPAL: OptionalConfigEntry[String] =
+    buildConf("celeborn.master.http.spnego.principal")
+      .categories("master")
+      .version("0.5.0")
+      .doc("SPNego service principal, typical value would look like HTTP/_HOST@EXAMPLE.COM." +
+        " SPNego service principal would be used when celeborn http authentication is enabled." +
+        " This needs to be set only if SPNEGO is to be used in authentication.")
+      .stringConf
+      .createOptional
+
+  val MASTER_HTTP_PROXY_CLIENT_IP_HEADER: ConfigEntry[String] =
+    buildConf("celeborn.master.http.proxy.client.ip.header")
+      .categories("master")
+      .doc("The HTTP header to record the real client IP address. If your server is behind a load" +
+        " balancer or other proxy, the server will see this load balancer or proxy IP address as" +
+        " the client IP address, to get around this common issue, most load balancers or proxies" +
+        " offer the ability to record the real remote IP address in an HTTP header that will be" +
+        " added to the request for other devices to use. Note that, because the header value can" +
+        " be specified to any IP address, so it will not be used for authentication.")
+      .version("0.5.0")
+      .stringConf
+      .createWithDefault("X-Real-IP")
+
+  val MASTER_HTTP_AUTH_BASIC_PROVIDER_CLASS: ConfigEntry[String] =
+    buildConf("celeborn.master.http.auth.basic.provider.class")
```

Comment:
let's remove `.class` and allow it to accept both the short name of the built-in implementations alias and user-provided 3-rd full class name
```suggestion
    buildConf("celeborn.master.http.auth.basic.provider")
```